crates/interfaces/src/test_utils/generators.rs

```diff
@@ -37,33 +38,23 @@ pub fn rng() -> StdRng {
 /// in the result will be equal to `head`.
 ///
 /// The headers are assumed to not be correct if validated.
-pub fn random_header_range<R: Rng>(
-    rng: &mut R,
+pub fn random_header_range<'a>(
+    u: &mut Unstructured<'a>,
     range: std::ops::Range<u64>,
     head: H256,
 ) -> Vec<SealedHeader> {
     let mut headers = Vec::with_capacity(range.end.saturating_sub(range.start) as usize);
     for idx in range {
-        headers.push(random_header(
-            rng,
-            idx,
-            Some(headers.last().map(|h: &SealedHeader| h.hash()).unwrap_or(head)),
-        ));
+        headers.push(random_header(u));
     }
     headers
 }
 
 /// Generate a random [SealedHeader].
 ///
 /// The header is assumed to not be correct if validated.
-pub fn random_header<R: Rng>(rng: &mut R, number: u64, parent: Option<H256>) -> SealedHeader {
-    let header = reth_primitives::Header {
-        number,
-        nonce: rng.gen(),
-        difficulty: U256::from(rng.gen::<u32>()),
-        parent_hash: parent.unwrap_or_default(),
-        ..Default::default()
-    };
+pub fn random_header<'a>(u: &mut Unstructured<'a>) -> SealedHeader {
```

Comment:
this needs to still take the parent hash and number to ensure it's a sequential chain of headers